sections/dashboard/Stake/InputCards/EscrowInputCard.tsx

```diff
@@ -1,112 +1,115 @@
 import { wei } from '@synthetixio/wei';
-import isNil from 'lodash/isNil';
 import { FC, useCallback, useMemo, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import styled from 'styled-components';
-import { useContractWrite, usePrepareContractWrite } from 'wagmi';
 
 import Button from 'components/Button';
 import NumericInput from 'components/Input/NumericInput';
 import SegmentedControl from 'components/SegmentedControl';
 import { DEFAULT_CRYPTO_DECIMALS, DEFAULT_TOKEN_DECIMALS } from 'constants/defaults';
-import { monitorTransaction } from 'contexts/RelayerContext';
-import { useStakingContext } from 'contexts/StakingContext';
-import { STAKING_LOW_GAS_LIMIT } from 'queries/staking/utils';
-import { truncateNumbers, zeroBN } from 'utils/formatters/number';
+import { useAppDispatch, useAppSelector } from 'state/hooks';
+import { approveKwentaToken, stakeEscrow, unstakeEscrow } from 'state/staking/actions';
+import {
+	selectEscrowedKwentaBalance,
+	selectIsKwentaTokenApproved,
+	selectStakedEscrowedKwentaBalance,
+} from 'state/staking/selectors';
+import { toWei, truncateNumbers, zeroBN } from 'utils/formatters/number';
 
 import { StakingCard } from '../common';
 
 const EscrowInputCard: FC = () => {
 	const { t } = useTranslation();
-	const {
-		stakedEscrowedBalance,
-		escrowedBalance,
-		kwentaApproveConfig,
-		kwentaTokenApproval,
-		rewardEscrowContract,
-	} = useStakingContext();
+	const dispatch = useAppDispatch();
 
 	const [amount, setAmount] = useState('');
 	const [activeTab, setActiveTab] = useState(0);
 
-	const amountBN = useMemo(
-		() => (amount === '' ? zeroBN.toString(0, true) : wei(amount).toString(0, true)),
-		[amount]
-	);
+	const escrowedKwentaBalance = useAppSelector(selectEscrowedKwentaBalance);
+	const stakedEscrowedKwentaBalance = useAppSelector(selectStakedEscrowedKwentaBalance);
+	const isKwentaTokenApproved = useAppSelector(selectIsKwentaTokenApproved);
+
+	const amountBN = useMemo(() => toWei(amount).toString(0, true), [amount]);
```

Comment:
named amountBN but it's returning a string. If it's for passing the calls as BigNumber wei value I think a better option could be  `wei(amount).toBN()` and we make those params BigNumber where it's more clear the expected type is a wei denomination.

Reply:
Done